### PR TITLE
Scripting: Whitelist API spec gradle task

### PR DESCRIPTION
**Pull request by stu-elastic**
Adds `generateContextApiSpec` gradle task that generates whitelist api
specs under `modules/lang-painless/src/main/generated/whitelist-json`.

The common classes are in `painless-common.json`, the specialized classes
per context are in `painless-$context.json`.

eg. `painless-aggs.json` has the specialization for the aggs contexts

### CODE REVIEW

**Comment by stu-elastic** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-unix

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextGeneratorCommon.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+                    compare = n1.compareTo(n2);
+                } else {
+                    compare = Boolean.compare(i1, i2) * -1;
+                }
+            }
+
+            return compare;
+        });
+
+        // TODO(stu): unmodifiable list
```
Comment (stu-elastic) [inline modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextGeneratorCommon.java:173]: TODO remove TODO

**Review by jdconrad — APPROVED** [review]
LGTM. Thank you for taking this on!

A couple of thoughts:
* We may want to look at some refactoring as you mentioned in our PR walkthough since this code is used in multiple places. 
* Now that there's multiple files I would consider adding a package for whitelist info generation.

**Comment by stu-elastic** [comment]
master (8.0): 2aa2224 
7.x (7.11): 5b8bac3

**Pull request merged**